app/src/main/java/org/simple/clinic/widgets/ageanddateofbirth/DateOfBirthFormatValidator.kt

```diff
@@ -17,27 +17,42 @@ class DateOfBirthFormatValidator @Inject constructor(
     DATE_IS_IN_FUTURE,
   }
 
+  sealed class Result2 {
+    data class Valid(val parsedDate: LocalDate) : Result2()
+    sealed class Invalid : Result2() {
+      object InvalidPattern : Invalid()
+      object DateIsInFuture : Invalid()
+    }
+  }
+
+  @Deprecated(
+      message = "use validate2 instead which uses sealed classes instead of enums",
+      replaceWith = ReplaceWith("validate2(dateText, nowDate)"))
   fun validate(dateText: String, nowDate: LocalDate = LocalDate.now(UTC)): Result {
-    return try {
+    return when (validate2(dateText, nowDate)) {
+      is Result2.Valid -> Result.VALID
+      is Result2.Invalid.InvalidPattern -> Result.INVALID_PATTERN
+      is Result2.Invalid.DateIsInFuture -> Result.DATE_IS_IN_FUTURE
+    }
+  }
+
+  fun validate2(dateText: String, nowDate: LocalDate = LocalDate.now(UTC)): Result2 {
```

Comment:
I'm not too sure we should be defaulting to the UTC date here. Since this is used for validation of user input, user will enter a date in local time, while here we'll compare it against the UTC time. This might lead to subtle bugs.

What do you think about either:
- Injecting the system time zone and use that to generate the `nowDate`.
- Removing the default parameter value and forcing users to provide the `nowDate` all the time?

Reply:
Good catch. Because the class is now called `UserInputDateValidator`, I think we can just inject the user's timezone. What do you think?

Reply:
Works for me.